services/s3/src/main/resources/codegen-resources/customization.config

```diff
@@ -159,5 +159,6 @@
     "createMethodParams": [
       "clientConfiguration"
     ]
-  }
+  },
+  "extensionInterface": "software.amazon.awssdk.services.s3.extensions.S3ClientExtensionMethods"
```

Comment:
When we eventually support async extensions as well, do you foresee having 2 fields here? Would it maybe be simpler to just have the package prefix, and we can infer the rest? Or even a simple boolean if we can infer all of it.

Reply:
I think a name is better, at least a package name; leads to less inference. When I added this, I had in mind that we just slap Async on, but that's quite ugly. We could either infer both names or add a more complex structure.